projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/src/serverRequests.ts

```diff
@@ -113,3 +113,73 @@ export async function jobdDataRequest(): Promise<void> {
     );
   }
 }
+
+/**
+ * Sent a POST request to the server to get more information about the notebook that is failing the job run
+ * Returns the path to the notebook file and the index of the cell that is failing in the notebook
+ */
+export async function getFailingNotebookInfo(failingCellId: string): Promise<{
+  path: string;
+  failingCellIndex: string;
+}> {
+  type getFailingNotebookInfoResult = {
+    path: string;
+    failingCellIndex: string;
+  };
+  const dataToSend = {
+    failingCellId: failingCellId,
+    jobPath: jobData.get(VdkOption.PATH)
+  };
+  if (await checkIfVdkOptionDataIsDefined(VdkOption.PATH)) {
+    try {
+      const data = await requestAPI<getFailingNotebookInfoResult>(
+        'failingNotebook',
+        {
+          body: JSON.stringify(dataToSend),
+          method: 'POST'
+        }
+      );
+      return {
+        path: data['path'],
+        failingCellIndex: data['failingCellIndex']
+      };
+    } catch (error) {
+      await showErrorMessage('Encountered an error. Error:', error, [
+        Dialog.okButton()
+      ]);
+      return {
+        path: '',
+        failingCellIndex: ''
+      };
+    }
+  } else {
+    return {
+      path: '',
+      failingCellIndex: ''
+    };
+  }
+}
+
+/**
+ * Sent a POST request to the server to indices of the vdk cells of a notebook
+ * Returns an Array with indices if vdk cells are found and empty array if not
+ */
+export async function getVdkCellIndices(
+  nbPath: string
+): Promise<Array<Number>> {
+  try {
+    const dataToSend = {
+      nbPath: nbPath
+    };
+    const data = await requestAPI<Array<Number>>('vdkIndices', {
+      body: JSON.stringify(dataToSend),
+      method: 'POST'
+    });
+    return data;
+  } catch (error) {
+    await showErrorMessage('Encountered an error. Error:', error, [
+      Dialog.okButton()
+    ]);
```

Comment:
nit: I can frequently see a similar code block:
```
await showErrorMessage('Encountered an error. Error:', error, [
      Dialog.okButton()
    ]);
```
May extract commons, to deduplicate

Reply:
but it is just 1 function would it get any better if we extract it?

Reply:
I extracted it into a different function